internal/pkg/eql/expression.go

```diff
@@ -62,13 +62,45 @@ func New(expression string) (*Expression, error) {
 		return nil, ErrEmptyExpression
 	}
 
+	errorListener := newErrorListener()
 	input := antlr.NewInputStream(expression)
 	lexer := parser.NewEqlLexer(input)
 	lexer.RemoveErrorListeners()
+	lexer.AddErrorListener(errorListener)
 	tokens := antlr.NewCommonTokenStream(lexer, antlr.TokenDefaultChannel)
 	p := parser.NewEqlParser(tokens)
 	p.RemoveErrorListeners()
+	p.AddErrorListener(errorListener)
 	tree := p.ExpList()
 
+	if len(errorListener.errors) > 0 {
+		// TODO: When we hit go 1.20 this should become
```

Comment:
can you create an issue or fix? having more errors up to N may be useful

Reply:
you can also use https://github.com/hashicorp/go-multierror we use it on multiple places